### PR TITLE
Raven db 13655

### PR DESCRIPTION
**Pull request by talweiss1982**

### CODE REVIEW

File: test/SlowTests/Issues/RavenDB_12151.cs [hunk 1]
```diff
@@ -116,7 +116,7 @@ public void IndexingWhenGlobalScratchSpaceLimitExceeded32()
             }
         }
 
-        [Fact]
+        [Fact(Skip ="Transaction size changed below 1MB")]
```
Comment (ayende) [inline test/SlowTests/Issues/RavenDB_12151.cs]: Why this change?

Comment (ayende) [reply test/SlowTests/Issues/RavenDB_12151.cs]: What changed the TX size? We need to fix the test / figure out why it changed

Comment (talweiss1982) [reply test/SlowTests/Issues/RavenDB_12151.cs]: hard to track what caused it, @maximburyak and I decided to finish fixing the rest of the tests and try to figure this one on a later PR since it was taking too long and I couldn't track what caused it.
I know for sure this test was passing when I wrote it.
now the largest Tx is 334kb...

Comment (talweiss1982) [reply test/SlowTests/Issues/RavenDB_12151.cs]: It is likely the limitations I have put on the merger that are limiting the indexing transactions. I'll open a separate issue to investigate this but for the short term let's get the tests green again.

Comment (talweiss1982) [reply test/SlowTests/Issues/RavenDB_12151.cs]: It is likely the limitations I have put on the merger that are limiting the indexing transactions. I'll open a separate issue to investigate this but for the short term let's get the tests green again.

**Review by arekpalinski — APPROVED** [review]

File: test/SlowTests/Issues/RavenDB_12151.cs (hunk 1 above)
Comment (maximburyak) [reply test/SlowTests/Issues/RavenDB_12151.cs]: That was resolved in the commit below

File: src/Raven.Server/Documents/Indexes/Index.cs [hunk 2]
```diff
@@ -3278,6 +3276,9 @@ private int MinBatchSize
         {
             var txAllocationsInBytes = UpdateThreadAllocations(indexingContext, indexWriteOperation, stats, updateReduceStats: false);
 
+            //We need to take the read transaction encryption size into account as we might read alot of document and produce very little indexing output.
+            txAllocationsInBytes += documentsOperationContext.Transaction.InnerTransaction.LowLevelTransaction.TotalEncryptionBufferSize.GetValue(SizeUnit.Bytes);
```
Comment (maximburyak) [inline src/Raven.Server/Documents/Indexes/Index.cs:3280]: Nice

File: src/Raven.Server/Documents/Replication/ReplicationLoader.cs [hunk 3]
```diff
@@ -1217,7 +1217,9 @@ public void Dispose()
                 }
             });
 
-            ea.Execute(() => ConflictResolver?.ResolveConflictsTask.Wait());
+            ea.Execute(() => ConflictResolver?.ResolveConflictsTask.Wait(TimeSpan.FromSeconds(15)));
```
Comment (maximburyak) [inline src/Raven.Server/Documents/Replication/ReplicationLoader.cs]: I'm not sure about the consequences of not waiting through here, and we do a similar unbounded wait few line above, although I agree that unbounded waits in dispose function is a bad practice in general.
```suggestion
            ea.Execute(() => ConflictResolver?.ResolveConflictsTask.Wait());
```

File: test/SlowTests/Client/Subscriptions/RavenDB-8404.cs [hunk 4]
```diff
@@ -48,7 +48,11 @@ public async Task SubscriptionsRQLSupportStartsWith()
                 _ = subscription.Run(x =>
                 {
                     users.AddRange(x.Items.Select(i => i.Result));
-                    amre.Set();
+                    //in 32bit we seem to send batches of 1 User
+                    if (users.Count == 3)
```
Comment (maximburyak) [inline test/SlowTests/Client/Subscriptions/RavenDB-8404.cs:52]: Right, because we started limiting the batch size in subscriptions

**Review by maximburyak — CHANGES_REQUESTED** [review]

**Review by maximburyak — APPROVED** [review]

**Review by ayende — APPROVED** [review]
Very nice work.
